test/client_and_receiver_integration_test.exs

```diff
@@ -262,39 +297,39 @@ defmodule SenderAndReceiverIntegrationTest do
     end
 
     @tag :tls
-    @tag port: 8154
+    @tag port: 8155
```

Comment:
Had to change port numbers all the way down. There's a better  way of handling this, namely using port 0 and providing the randomly chosen port to the test. This IMO should  be done in another PR.

Reply:
correct, I also didn't like using so many ports. I tried using only one port but it fails when parallel tests are running in CI. Locally it worked fine

Reply:
Interesting, I'll give it a second set of eyes later today, may simply be not possible in CI, but we could also just make a  custom random port function in tests ourselves (i.e., get all listening ports, pick  a  port number not being used).